cryptoprotocol/src/main/java/de/upb/crypto/incentive/cryptoprotocol/model/User.java

```diff
@@ -0,0 +1,64 @@
+package de.upb.crypto.incentive.cryptoprotocol.model;
+
+import de.upb.crypto.incentive.cryptoprotocol.model.keys.provider.ProviderPublicKey;
+import de.upb.crypto.incentive.cryptoprotocol.model.keys.user.UserKeyPair;
+import de.upb.crypto.math.serialization.StringRepresentation;
+import de.upb.crypto.math.serialization.annotations.ReprUtil;
+import de.upb.crypto.math.structures.rings.zn.Zn;
+import de.upb.crypto.math.structures.rings.zn.Zn.ZnElement;
+import lombok.Data;
+
+@Data
+public class User
+{
+    private UserKeyPair userKeyPair;
+    private Token userToken;
+
+    /**
+     * (effectively a) no args constructor, generating new key pair upon instantiation. Used when new session in app is started.
+     */
+    public User(PublicParameters pp, ProviderPublicKey pk)
```

Comment:
Written like this I have to create a new user Object to get a new token. We would like to have many tokens, one for each "campaign"

Reply:
We can also discuss if the Token class would be able to "generate" a new token in the form that you have written here, i.e. everything getting a value from `getUniformlyRandomElement();`.

Reply:
Changed it to be an immutable data class.